### PR TITLE
CORS-3517: Add the GCP CAPI periodic test

### PR DESCRIPTION
**Pull request by barbacbd**

### CODE REVIEW

File: ci-operator/config/openshift/installer/openshift-installer-master__altinfra-periodics.yaml [hunk 1]
```diff
@@ -106,6 +106,15 @@ tests:
         keyC valueC
     workflow: openshift-e2e-aws-sharednetwork
   timeout: 6h0m0s
+- as: e2e-gcp-capi-ovn
+  interval: 8h
+  steps:
+    cluster_profile: gcp
+    env:
+      FEATURE_GATES: '["ClusterAPIInstall=true"]'
+      FEATURE_SET: CustomNoUpgrade
```
Comment (patrickdillon) [inline ci-operator/config/openshift/installer/openshift-installer-master__altinfra-periodics.yaml:115]: After discussion late last week with the TRT team, I understood that we want to favor testing the TechPreview feature set--not using individual gates like this. So for the goal of this PR, you should check whether GCP Tech Preview tests are present in the release periodics: 
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/release/openshift-release-master__nightly-4.16.yaml
and 
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/release/openshift-release-master__ci-4.16.yaml

It looks like GCP techpreview jobs are in the ci periodics, but not the nightly periodics. So I believe we want to add the gcp periodics to the nightly config. AWS is also missing from the nightly config, so if you can add that at the same time that would be appreciated.

**Comment by patrickdillon** [comment]
The tech preview periodics should go into the openshift/release config. See https://github.com/openshift/release/pull/51731/files for examples

Soon we can remove this entire altinfra variant

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.16-altinfra-e2e-gcp-capi-ovn

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.17-altinfra-e2e-gcp-ovn-byo-network-capi
/pj-rehearse pull-ci-openshift-installer-release-4.17-altinfra-e2e-gcp-ovn-xpn-capi

**Comment by barbacbd** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.17-altinfra-e2e-gcp-ovn-xpn-capi

**Comment by patrickdillon** [comment]
I'm not sure we need this... at least the tech preview periodic, as we already have gcp techpreview periodics: https://sippy.dptools.openshift.org/sippy-ng/jobs/4.17/analysis?filters=%7B%22items%22%3A%5B%7B%22columnField%22%3A%22name%22%2C%22operatorValue%22%3A%22equals%22%2C%22value%22%3A%22periodic-ci-openshift-release-master-ci-4.17-e2e-gcp-ovn-techpreview-serial%22%7D%5D%7D 

And I'm not sure I see much point in changing the altinfra job to be techpreview rather than customnoupgrade. I'm open to it, but it will introduce more instability as it enables other techpreview features. :thinking:

**Pull request closed without merging**